src/convertToObject.js

```diff
@@ -5,8 +5,48 @@
  *
  * @return {object}
  */
-function convertToObject(sourceString) {
-  // write your code here
-}
+const convertToObject = function (sourceString) {
+  const styleObject = {};
+
+  const cleanSourceString = sourceString
+    .replace(/[\t\r]+/g, '')
+    .replace(/\s*;\s*/g, ';')
+    .replace(/\s*:\s*/g, ':')
+    .replace(/;\n/g, ';');
+
+  const declarationsArray = cleanSourceString.trim().split(';').filter(Boolean);
+
+  let buffer = '';
+  let insideMultilineValue = false;
+
+  declarationsArray.forEach(function (declarationItem) {
+    if (insideMultilineValue) {
+      buffer += declarationItem.trim();
+
+      if (!/\\$/.test(buffer)) {
+        const parts = buffer.split(':');
+
+        if (parts.length === 2) {
+          styleObject[parts[0].trim()] = parts[1].trim();
+        }
+        insideMultilineValue = false;
+        buffer = '';
+      }
+    } else {
+      const parts = declarationItem.split(':');
+
+      if (parts.length === 2) {
+        if (/\\$/.test(parts[1].trim())) {
+          buffer = declarationItem.trim();
+          insideMultilineValue = true;
+        } else {
+          styleObject[parts[0].trim()] = parts[1].trim();
+        }
+      }
+    }
```

Comment:
try to simplify this logic using the reduce method, feel free to ask for some help in the chat